### PR TITLE
add means to sort objects

### LINKED ISSUE #27: sorted dicts
i find it useful to sort dicts when displaying them:  `using DataStructures; eye(SortedDict(mydict))`.  the problem is that only works at the top level, and my trees typically have nested dicts.  would be nice to have a menu option to sort a node that is a dict.  happy to help if you point me in the right direction.

### PR DESCRIPTION
**Pull request by bjarthur**
closes https://github.com/tshort/Eyeball.jl/issues/27

### CODE REVIEW

File: src/Eyeball.jl [hunk 1]
```diff
@@ -162,6 +162,12 @@ function eye(x = Main, depth = 10; interactive = true, all = false)
             sobj = String(take!(io.io))
             _pager(node.data.value)
             resetterm()
+        elseif i == Int('S')
+            node = FoldingTrees.setcurrent!(menu, menu.cursoridx)
+            length(methods(sort, (typeof(node.data.value),)))==0 && return false
```
Comment (tshort) [inline src/Eyeball.jl]: Is this the best way to check if an object can be sorted? It may be. It just looks clunky.

Comment (bjarthur) [reply src/Eyeball.jl]: the only other way i can think of is a try-catch block, but that's even worse.  happy to make a change if you think of sth better.

**Comment by bjarthur** [comment]
the help menu is not displayed after quitting from a sort:

```
julia> eye(Dict(1=>'a', 2=>'b', 3=>Dict(4=>'d', 5=>'e')))
[f] fields [d] docs [e] expand [m/M] methodswith [o] open [r] tree [s] show [S] Sort [t] typeof [z] summarize [q] quit
 >   : Dict{Int64, Any}  Dict{Int64, Any}(2=>'b', 3=>Dict(5=>'e', 4=>'d'), 1=>'a')
      2: Char  'b'
      3: Dict{Int64, Char}  Dict(5=>'e', 4=>'d')
       5: Char  'e'
       4: Char  'd'
      1: Char  'a'

Opening sort(``) ...

[f] fields [d] docs [e] expand [m/M] methodswith [o] open [r] tree [s] show [S] Sort [t] typeof [z] summarize [q] quit
 >   : OrderedDict{Int64, Any}  OrderedDict{Int64, Any}(1=>'a', 2=>'b', 3=>Dict(5=>'e', 4=>'d'))
      1: Char  'a'
 >   : Dict{Int64, Any}  Dict{Int64, Any}(2=>'b', 3=>Dict(5=>'e', 4=>'d'), 1=>'a')
      2: Char  'b'
      3: Dict{Int64, Char}  Dict(5=>'e', 4=>'d')
       5: Char  'e'
       4: Char  'd'
      1: Char  'a'
```

**Comment by bjarthur** [comment]
also, maybe it would be better to sort in-place, without opening a new tree at the current node?

**Comment by tshort** [comment]
Yeah, but in place is harder to implement.

File: src/Eyeball.jl (hunk 1 above)
Comment (tshort) [reply src/Eyeball.jl]: How about `hasmethod`?

Comment (bjarthur) [reply src/Eyeball.jl]: changed!

**Comment by bjarthur** [comment]
what does 'O' do?  i don't see any diff in output with 'o' for my simple test cases.  and it's not documented.  the code has `All`...

**Comment by tshort** [comment]
@bjarthur, this is good to go, right?

**Comment by tshort** [comment]
> what does 'O' do? i don't see any diff in output with 'o' for my simple test cases. and it's not documented. the code has `All`...

Right now, it opens modules with `names(y, all = true)` (includes non-exported names defined in the module, deprecated names, and compiler-generated names).  It could be extended to other types where you want to show more content.

**Comment by bjarthur** [comment]
yes, it's ready!  and i just noticed `O` is doc'd in the readme, just not in the repl.

**Comment by tshort** [comment]
> the help menu is not displayed after quitting from a sort:

This happens any time you open another view. I haven't looked into a fix.

**Pull request merged**